src/transformers/modeling_lxmert.py

```diff
@@ -0,0 +1,975 @@
+        embedding_output = self.embeddings(input_ids, token_type_ids)
+
+        # Run Lxmert encoder
+        x_encoder_outputs = self.encoder(
+            embedding_output,
+            extended_attention_mask,
+            visn_feats=visual_feats,
+            visn_attention_mask=extended_visual_attention_mask,
+            output_v_attentions=output_v_attentions,
+            output_l_attentions=output_l_attentions,
+            output_x_attentions=output_x_attentions,
+        )
+
+        v_outputs, l_outputs = x_encoder_outputs[0], x_encoder_outputs[1]
+        visn_output = v_outputs[0][-1]
+        lang_output = l_outputs[0][-1]
+        pooled_output = self.pooler(lang_output)
+
+        if not return_dict:
+            return (
+                (lang_output, visn_output, pooled_output),
+                (v_outputs[0][:-1], l_outputs[0][:-1]),
+                (
+                    v_outputs[1:] if output_v_attentions else None,
+                    l_outputs[1:] if output_l_attentions else None,
+                    x_encoder_outputs[2:] if output_x_attentions else None,
+                ),
+            )
+
+        return LxmertModelOutput(
+            last_hidden_state_l=lang_output,
+            hidden_states_l=l_outputs[0][:-1],
+            attentions_l_encoder=output_l_attentions,
+            last_hidden_state_v=visn_output,
+            hidden_states_v=v_outputs[0][:-1],
+            attentions_v_encoder=output_v_attentions,
+            pooled_output_x_encoder=pooled_output,
+            attentions_x_encoder=output_x_attentions,
+        )
+
+
+@add_start_docstrings(
+    """Lxmert Model with a specified pre-training heads on top. """, LXMERT_START_DOCSTRING, LXMERT_INPUTS_DOCSTRING
+)
+class LxmertForPretraining(LxmertPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        # Configuration
+        self.config = config
+        self.num_answers = config.n_qa_labels
+        self.visual_loss_normalizer = config.visual_loss_normalizer
+
+        # Use of pre-training tasks
+        self.task_mask_lm = config.task_mask_lm
+        self.task_obj_predict = config.task_obj_predict
+        self.task_matched = config.task_matched
+        self.task_qa = config.task_qa
+
+        # Lxmert backbone
+        self.bert = LxmertModel(config)
+
+        # Pre-training heads
+        self.cls = LxmertPreTrainingHeads(config, self.bert.embeddings.word_embeddings.weight)
+        if self.task_obj_predict:
+            self.obj_predict_head = LxmertVisualObjHead(config)
+        if self.task_qa:
+            self.answer_head = LxmertVisualAnswerHead(config, self.num_answers)
+
+        # Weight initialization
+        self.init_weights()
+
+        # Loss functions
+        self.loss_fcts = {
+            "l2": SmoothL1Loss(reduction="none"),
+            "visn_ce": CrossEntropyLoss(ignore_index=-1, reduction="none"),
+            "ce": CrossEntropyLoss(ignore_index=-1),
```

Comment:
Previously we have `ignore_index=-1` but it brought all kinds of problems and mismatch betcause it wasn't PyTorch's default. We now try to have `ignore_index=-100` (the default) everywhere.

Would it be possible to do the same here?
```suggestion
            "visn_ce": CrossEntropyLoss(reduction="none"),
            "ce": CrossEntropyLoss(),
```

Reply:
I think for testing the local data, switching to -100 would lead to some problems for the local data I am using for testing, but for the future, I think once we implement a vision-language pipeline it would make much more sense, so yes I can definitely change the ignore index for this PR!